### PR TITLE
task done

### PR DESCRIPTION
**Pull request by RamanovskiyVV**

### CODE REVIEW

File: src/app/features/courses/courses.html [hunk 1]
```diff
@@ -1,5 +1,11 @@
-<header common-header></header>
-<card *ngFor="let course of listOfCourses; trackBy:identify" title="{{course.title}}" text="{{course.description}}"
-    creationDate="{{course.creationDate}}" duration="{{course.duration}}" authors="{{course.authors}}"
-    editable={{course.editable}}>
+<search></search>
+<card
+    *ngFor="let course of listOfCourses; trackBy:identify"
+    title="{{course.title}}"
```
Comment (litwin90) [inline src/app/features/courses/courses.html]: usually we use such a syntax: 
[title]="course.title"

File: src/app/features/courses/courses.module.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
 import { SharedModule } from 'src/app/shared/shared.module';
 import { CourseCardComponent } from './courseCard.component';
-import { CoursesComponent } from './courses.component';
 
 @NgModule({
     imports: [SharedModule],
-    declarations: [CoursesComponent, CourseCardComponent],
-    exports: [CoursesComponent, CourseCardComponent],
+    declarations: [CourseCardComponent],
+    exports: [CourseCardComponent],
```
Comment (litwin90) [inline src/app/features/courses/courses.module.ts:8]: lets use ChangeDetection Onpush for all the components

File: src/app/features/courses/edit-course/edit-course.component.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { authorValidator } from 'src/app/shared/validators/author-validator';
+
+@Component({
+    selector: 'edit-course',
+    styleUrls: ['./edit-course.component.css'],
+    templateUrl: './edit-course.component.html',
```
Comment (litwin90) [inline src/app/features/courses/edit-course/edit-course.component.ts:9]: onPush

File: src/app/features/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Component, Input } from '@angular/core';
+import { User } from 'src/app/models/user.medel';
+
+@Component({
+    selector: 'login',
+    styleUrls: ['./login.component.css'],
+    templateUrl: './login.component.html',
```
Comment (litwin90) [inline src/app/features/login/login.component.ts:8]: onpush

File: src/app/features/registration/registration.component.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { emailValidator } from 'src/app/shared/validators/email-validator';
+
+@Component({
+    selector: 'registration',
+    styleUrls: ['./registration.component.css'],
+    templateUrl: './registration.component.html',
```
Comment (litwin90) [inline src/app/features/registration/registration.component.ts:9]: onPush

File: src/app/shared/components/search/search.component.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { Component, Input, Output } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+
+@Component({
+    selector: 'search',
+    styleUrls: ['./search.component.css'],
+    templateUrl: './search.component.html',
```
Comment (litwin90) [inline src/app/shared/components/search/search.component.ts:8]: onpush

File: src/app/shared/pipes/string-joiner.pipe.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({ name: 'string-joiner' })
```
Comment (litwin90) [inline src/app/shared/pipes/string-joiner.pipe.ts:3]: image

Comment (RamanovskiyVV) [reply src/app/shared/pipes/string-joiner.pipe.ts:3]: done

File: src/app/features/courses/courses.module.ts (hunk 2 above)
Comment (RamanovskiyVV) [reply src/app/features/courses/courses.module.ts:8]: done

**Pull request merged**